### DISCUSSION TITLE
targeting on Tags

### DISCUSSION THREAD

**Discussion by jehrlich** [apideck-libraries/portman#410] [Q&A] [answered]
It would be handy if Portman targeting could be on Tags in addition to operationId and method+path. In practice it seems that Tags are often used for grouping operations and so a natural discriminator for test groups.

BTW, I love this. Thank you.

**Comment by thim81** [comment, marked as answer]
hi @jehrlich,

No fully sure, what your use-case is?
Is it to execute only certain Postman requests with Tags via Newman? Or only want create a collection with requests from certain tags?

**Newman use-case:**
You can tell to the Newman CLI which "folders" (tags in OpenAPI) to execute.

Example `newman-options.json`
```
{
  "folder": "store"
  "environment": "./tmp/crm/postman-dev.env.json",
  "iteration-count": 5,
  "ignore-redirects": true,
  "insecure": true
}
```

Example:

```ssh
portman -u https://specs.apideck.com/crm.yml -c ./tmp/crm/portman-config.json --runNewman --newmanOptionsFile ./tmp/crm/newman-options.json
```


**Filtered collection use-case:**
Portman includes [openapi-format](https://github.com/thim81/openapi-format) which allows easy filtering options.

You can create an `openapi-filter.yaml` file with the tags (or [any of the other filter options](https://github.com/thim81/openapi-format#openapi-filter-options))

```yaml
tags:
  - store
  - user
```
Example:

```ssh
portman -u https://specs.apideck.com/crm.yml --filterFile openapi-filter.yaml 
```

**Comment by jehrlich** [comment]
That answers my use case. Thanks.

**Reply by jehrlich** [reply]
I'm still working on a github workflow intended to verify a public API documented in the repo. The API serves several classes of users, and endpoints are tagged according to class(es) of users. Authorization is via a header JWT token which identifies the user and user class. Therefore the first step in a run is to create a token using credentials for a test user of a particular class. Using Newman filtering, I could set up a run for each user class with different user credentials in each run.

It seems that if Portman allowed me to `assignVariables` or `extendTests` targeting on Tag values then the run could fetch tokens for each user class and use the appropriate token as needed.
